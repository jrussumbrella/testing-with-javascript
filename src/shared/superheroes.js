const superheroes = [
  {
    name: 'Batman',
    powers: ['rant', 'fly'],
  },
  {
    name: 'Superman',
    powers: ['rant'],
  },
]

export const getFlyingHeros = () => {
  return superheroes.filter(hero => hero.powers.includes('fly'))
}
