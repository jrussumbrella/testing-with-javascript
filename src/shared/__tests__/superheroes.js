import {getFlyingHeros} from '../superheroes'

test('return super heroes that fly', () => {
  const flyingHeroes = getFlyingHeros()
  expect(flyingHeroes).toMatchSnapshot()
})
