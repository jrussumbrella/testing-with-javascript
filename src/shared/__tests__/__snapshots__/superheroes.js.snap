// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return super heroes that fly 1`] = `
Array [
  Object {
    "name": "Batman",
    "powers": Array [
      "rant",
      "fly",
    ],
  },
]
`;
